turn_rasa_connector/test_turn.py

```diff
@@ -695,6 +735,72 @@ async def test_send_image_message(turn_mock_server: Sanic):
     assert len(turn_mock_server.app.media) == 1
 
 
+@pytest.mark.asyncio
+async def test_send_document_message(turn_mock_server: Sanic):
+    """
+    Makes an HTTP request to Turn to send the message
+    """
+    output_channel = TurnOutput(
+        url=f"http://{turn_mock_server.host}:{turn_mock_server.port}", token="testtoken"
+    )
+    await output_channel.send_response(
+        "27820001001",
+        {
+            "document": f"http://{turn_mock_server.host}:{turn_mock_server.port}"
+            "/documents/document.pdf",
+            "text": "test caption",
+        },
+    )
+    [message] = turn_mock_server.app.messages
+    assert message.json == {
+        "to": "27820001001",
+        "type": "document",
+        "document": {
+            "id": "d66084f148673c1abdcfdeeea673f2fb",
+            "caption": "test caption",
+        },
+    }
+    assert message.headers["Authorization"] == "Bearer testtoken"
+    assert message.headers["Content-Type"] == "application/json"
+
+    [media] = turn_mock_server.app.media
+    assert media.body == b"testdocumentcontent"
+    assert media.headers["Content-Type"] == "application/pdf"
+
+    await output_channel.send_response(
+        "27820001001",
+        {
+            "document": f"http://{turn_mock_server.host}:{turn_mock_server.port}"
+            "/documents/doucment.pdf"
```

Comment:
s/doucment/document? That should then hit to cache on this second request, so that we only make a single request to media server and reuse the media ID